src/sql/base/browser/ui/table/table.ts

```diff
@@ -111,9 +115,38 @@ export class Table<T extends Slick.SlickData> extends Widget implements IDisposa
 
 		this.mapMouseEvent(this._grid.onContextMenu, this._onContextMenu);
 		this.mapMouseEvent(this._grid.onClick, this._onClick);
+		this.mapMouseEvent(this._grid.onHeaderClick, this._onHeaderClick);
 		this._grid.onColumnsResized.subscribe(() => this._onColumnResize.fire());
 	}
 
+	private invalidateRange(start: number, end: number): void {
+		let refreshedRows = range(start, end);
+		this._grid.invalidateRows(refreshedRows, true);
+		this._grid.render();
+	}
+
+	private renderGridRange(startIndex: number, count: number): void {
+		let editor = this._grid.getCellEditor();
+		let oldValue = editor ? (<Slick.Editors.Text<T>>editor).getValue() : undefined;
+		let wasValueChanged = editor ? editor.isValueChanged() : false;
+		this.invalidateRange(startIndex, startIndex + count);
+		let activeCell = this.activeCell;
+		if (editor && activeCell.row >= startIndex && activeCell.row < startIndex + count) {
+			if (oldValue && wasValueChanged) {
+				(<Slick.Editors.Text<T>>editor).setValue(oldValue);
+			}
+		}
+	}
+
+	public rerenderGrid(start: number, end: number) {
+		//render grid range seems to have problems loading.
```

Comment:
what is the comment mean?  could you please elaborate on the problems and how they're being mitigated?

Reply:
This comment was used to indicate that renderGridRange, when the cell is being loaded for the first time, the cellEditor for the grid is null,